### ISSUE TITLE
AngularJS support has officially ended

### ISSUE THREAD

**Issue by Justman10000** [owncloud/music#1081]
When installing the dependencies, this was noticed:

> For the actively supported Angular, see https://www.npmjs.com/package/@angular/core. AngularJS support has officially ended. For extended AngularJS support options, see https://goo.gle/angularjs-path-forward

Maybe replace times 🤔?

**Comment by nicekiwi** [comment]
> Ideally, yes. But the thing is, migration from AngularJS to the modern Angular is a huge task, as the two frameworks are quite different. This would be almost like a total rewrite of the front-end. I expect the total amount of development work required to be several man-months.
> 
> I have taken the first steps on this path by converting some of the Javascript files to TypeScript. But even converting all the JS to TS is still a long way to go and that's just the step 1 of the Angular-migration. The time available for a hobby project is always limited, and some other topics on the app are usually more interesting than the framework update. Hence, I'm not certain that this migration gets done before Google decides to deprecate also the Angular framework.

Curiously I'm in a similar situation, having recently setup NextCloud and Music and also working for a company with a much larger AngularJS codebase that we're desperatly trying to leave. 😅 

Unfortunatly I've not come accross any examples of large projects doing the migration successfully at scale and any resources previously available to assist with it have since been retired, so a total re-write and split migration seems the only way foward, to either React or Angular. 

As it is I'm in the process of migrating our own codebase to TypeScript and would be interested in helping that transistion here too. Where we go form there I'm not sure.

**Comment by paulijar** [comment]
> As it is I'm in the process of migrating our own codebase to TypeScript and would be interested in helping that transistion here too. 

Nice, pull requests are welcome :). If you are going to submit any PR, then it's probably better to do this in multiple small PRs, migrating just a few files at a time, than in one huge one.

>Where we go form there I'm not sure.

Yeah, the JS to TS conversion is the easy part and then the real work just starts. I *think* that the next step should be to reorganize the code so that all the sources (TS, CSS, HTML template) related to one module would always be in one folder, as opposed to the current division where all the CSS is in one folder and all the templates in another etc. And after that, make sure that all the modules use isolated scopes; at the moment, some of them do and others don't. When all this is done, then it might be possible to start replacing the AngualrJS modules with Angular modules.

All the above is based on just some quick guides I have read online. I have no real-life experience on making such migration and only limited experience on the modern Angular in general. What worries me the most is that our current code base has lot of "magic" built-in over the years to keep the UI responsive and at least somewhat usable also on gigantic music collections of tens of thousands of songs. I have no idea if and how the same things can be achieved on the modern Angular.

**Comment by Justman10000** [comment]
> then it's probably better to do this in multiple small PRs, migrating just a few files at a time, than in one huge one.

Not really! Then it is more unclear and one must go into each PR individually to merge these individually! Furthermore, this would be the purest PR ID waste!

**Comment by nicekiwi** [comment]
> > then it's probably better to do this in multiple small PRs, migrating just a few files at a time, than in one huge one.
> 
> Not really! Then it is more unclear and one must go into each PR individually to merge these individually! Furthermore, this would be the purest PR ID waste!

This is not just for the sake of it, but to make reviewing and testing the PR changes much easier, especially for a volunteer project where not much time may be available to review. 

Multiple small focussed PRs is the way to go.
